pkg/kube/discover.go

```diff
@@ -44,9 +44,14 @@ func IsOSRouteGroupAvailable(ctx context.Context, cli discovery.DiscoveryInterfa
 
 // IsResAvailableInGroupVersion takes a resource and checks if that exists in the passed group and version
 func IsResAvailableInGroupVersion(ctx context.Context, cli discovery.DiscoveryInterface, groupName, version, resource string) (bool, error) {
+	// This call is going to fail with error type `*discovery.ErrGroupDiscoveryFailed` if there are
+	// some api-resources that are served by agg. API server and the agg API server is not ready.
```

Comment:
```suggestion
	// some api-resources that are served by aggregated API server and the agg API server is not ready.
```
[nit] mentioning the full word once will help new folks understand this better